Bugfix/fixes restart and wrong numbers

raftx24 opened the pull request.

raftx24 committed:
fixes wrong number with restart

aocneanu committed:
Apply fixes from StyleCI

raftx24 committed:
adds error in console mode

raftx24 committed:
Merge remote-tracking branch 'origin/bugfix/fixes-restart-and-wrong-numbers' into bugfix/fixes-restart-and-wrong-numbers

aocneanu committed:
Apply fixes from StyleCI

aocneanu commented on src/Enums/ImportType.php:
```diff
@@ -0,0 +1,11 @@
+
+use LaravelEnso\Enums\Services\Enum;
+
+class ImportType extends Enum
```
Types should be enough

raftx24 replied:
we don't need to type anymore

aocneanu commented on src/Models/DataImport.php:
```diff
@@ -210,4 +212,16 @@ public function import(?string $sheet = null)
+    {
+        $this->when(
+            $this->template()->type() === ImportType::Insert,
+            fn () => $this->fill(['failed' => -1 * $this->successful]),
```
this will produce a lot of awkward situations. Imagine failed imports with -233 failed.

I could live with resetting both to 0 on all cases

aocneanu committed:
Update Chunk.php

aocneanu commented on src/Services/Template.php:
```diff
@@ -37,6 +38,13 @@ public function timeout(): int
+        return $this->template->has('type')
+            ? $this->template->get('type')
+            : ImportType::Insert;
+    }
```
we should also validate the type in template validation

raftx24 replied:
we don't need to type anymore

raftx24 committed:
addresses reviews.
reset counters to zero when data-import was restarted in all cases.
removes ImportTypes.

raftx24 committed:
Merge remote-tracking branch 'origin/bugfix/fixes-restart-and-wrong-numbers' into bugfix/fixes-restart-and-wrong-numbers

aocneanu committed:
Apply fixes from StyleCI
